### ISSUE TITLE
react-navigation@next does not show header with transparentModal

### ISSUE THREAD

**Issue by Aryk** [react-navigation/navigation-ex#240]
I was using createNativeStackNavigator to create a new stack. For one of the screens when setting presentation to "transparentModal" or "modal" in my iOS simulator, the overlay comes up but the header is completely gone. Removing {presentation: "transparentModal"} defaults to the push method and it shows the header like its supposed to.

However, with"transparentModal" or "modal" if the screen is another stack, it will show the header in that stack correctly. Is this normal functionality for modals?

**Comment by Hirbod** [comment]
The header will appear when the modal has its own Stack (nesting another Stack inside) but currently Android will break but it will work perfectly with iOS. (it does need react-native-screens > alpha.12, because nesting was fixed in 13. So when you're on EXPO like me, you can't use it)

**Comment by Aryk** [comment]
I see what you mean. I'm using another stack inside and I get the header...but there is no back button to remove the new modal because it is a new stack. So I have to set header to false anyways, and built the header manually so I can add the left back button.

Also with "modal", on iOS it does not go all the way up to the top of the screen. Is there anyway to control that?

**Comment by Hirbod** [comment]
It does not go all the way top, because its the new iOS 13 PageSheet style. You can change "presentation" to "transparentModal", which implements the old fullscreen view.

Maybe you can force to show the back button
https://reactnavigation.org/docs/en/next/native-stack-navigator.html

using headerBack visible props. The original implementation from react-native-screens also have a `headerLeft` which is missing here currently (only headerRight available)

**Comment by Aryk** [comment]
`headerBack` (and `headerLeft`) are both missing from the documentation on that page and I tried it. It does not work. There is no way to control the left side *fully* on a NativeStackNavigator currently atleast.

**Comment by Hirbod** [comment]
Yeah, as said, it's not implemented yet, but the underlying native library (react-native-screens) has already support for it (buggy though), so it's a feature request for nav-ex i guess.

**Comment by Hirbod** [comment]
https://github.com/kmagiera/react-native-screens/pull/260 there is a PR addressing this at upstream.
